File: Examples/VisualBasic/NoteBook/RemoveChildNode.vb
Imports System.IO
Imports Aspose.Note
Imports System.Collections.Generic
Imports AsposeNote = Aspose.Note

Namespace NoteBook
    Public Class RemoveChildNode
        Public Shared Sub Run()
            ' ExStart:RemoveChildNode
            ' The path to the documents directory.
            Dim dataDir As String = RunExamples.GetDataDir_NoteBook()

            ' Load a OneNote Notebook
            Dim notebook = New AsposeNote.Notebook(dataDir & Convert.ToString("test.onetoc2"))

            ' Traverse through its child nodes for searching the desired child item
            For Each child In New List(Of INotebookChildNode)(notebook)
                If child.DisplayName = "Remove Me" Then
                    ' Remove the Child Item from the Notebook
                    notebook.RemoveChild(child)
                End If
            Next

            dataDir = dataDir & Convert.ToString("RemoveChildNode_out_.onetoc2")

            ' Save the Notebook
            notebook.Save(dataDir)
            ' ExEnd:RemoveChildNode

            Console.WriteLine(Convert.ToString(vbLf & "Child node removed successfully." & vbLf & "File saved at ") & dataDir)
        End Sub
    End Class
End Namespace

File: Examples/CSharp/NoteBook/RemoveChildNode.cs
using System.IO;
using Aspose.Note;
using System;
using System.Collections.Generic;

namespace Aspose.Note.Examples.CSharp.WorkingWithNoteBook
{
    public class RemoveChildNode
    {
        public static void Run()
        {
            // ExStart:RemoveChildNode
            // ExFor:Notebook
            // ExFor:Notebook.DisplayName
            // ExFor:Notebook.RemoveChild
            // ExSummary:Shows how to remove a section from a notebook.

            // The path to the documents directory.
            string dataDir = RunExamples.GetDataDir_NoteBook();

            // Load a OneNote Notebook
            var notebook = new Notebook(dataDir + "test.onetoc2");

            // Traverse through its child nodes for searching the desired child item
            foreach (var child in new List<INotebookChildNode>(notebook))
            {
                if (child.DisplayName == "Remove Me")
                {
                    // Remove the Child Item from the Notebook
                    notebook.RemoveChild(child);
                }
            }

            dataDir = dataDir + "RemoveChildNode_out.onetoc2";

            // Save the Notebook
            notebook.Save(dataDir);

            // ExEnd:RemoveChildNode

            Console.WriteLine("\nChild node removed successfully.\nFile saved at " + dataDir);
        }
    }
}